### PR TITLE
Password reset docs and script

### PR DESCRIPTION
**Pull request by richvdh**
Replace the bash/perl gen_password script with a python one, and write a note
on how to use it.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -525,6 +525,26 @@ Logging In To An Existing Account
 Just enter the ``@localpart:my.domain.here`` Matrix user ID and password into
 the form and click the Login button.
 
+Password reset
+==============
+
+Synapse does not yet support a password-reset function (see
+https://matrix.org/jira/browse/SYN-11). In the meantime, it is possible to
+manually reset a user's password via direct database access.
```
Comment (erikjohnston) [inline README.rst]: Can you explicitly spell out that we do have one if the user has specified an email address please?

File: scripts/hash_password [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+
+import argparse
+import bcrypt
+import getpass
+
+bcrypt_rounds=12
```
Comment (erikjohnston) [inline scripts/hash_password]: What happens if we use a different number of rounds? The number of rounds can be configured in the config iirc.

File: README.rst (hunk 1 above)
Comment (richvdh) [inline README.rst]: Oh, we do? I hadn't realised that.

File: scripts/hash_password (hunk 2 above)
Comment (richvdh) [inline scripts/hash_password]: the number of rounds is recorded in the hash (`$2a$12$xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx` -> salt is 2a, # of rounds is 12) so synapse will be happy with the generated hash either way.

Comment (erikjohnston) [inline scripts/hash_password]: That makes me happy :)

**Comment by richvdh** [comment]
@erikjohnston ptal

**Comment by erikjohnston** [comment]
Ta, LGTM

File: scripts/hash_password (hunk 2 above)
Comment (richvdh) [inline scripts/hash_password]: actually 2a describes the hashing algorithm rather than the salt

**Pull request merged**